README.md
# Instabot

An instagram bot written in Python

changes.txt
1. Updated xpath element for like button (Under invalid like element in github issues)
2. Updated the webdriverdownloader.py @ line 320 to account for changes within /home/todd/.local/share/virtualenvs/Instabot-jvzHF1RQ/lib/python3.7/site-packages
New changes: filename = [name for name in filenames if os_name + bitness in name and name[-3:] != 'asc']

app.py
from instapy import InstaPy

session = InstaPy(username="", password='', headless_browser=True).login()
session.set_quota_supervisor(enabled=True, sleep_after=["likes", "comments_d", "follows", "unfollows", "server_calls_h"], sleepyhead=True, stochastic_flow=True, notify_me=False,
                              peak_likes_hourly=20,
                              peak_likes_daily=100,
                               peak_comments_hourly=12,
                               peak_comments_daily=180,
                                peak_follows_hourly=20,
                                peak_follows_daily=150,
                                 peak_unfollows_hourly=10,
                                 peak_unfollows_daily=50,
                                  peak_server_calls_hourly=None,
                                  peak_server_calls_daily=4000)

session.set_use_clarifai(enabled=True, api_key='')
session.set_do_like(enabled=False)
session.set_do_comment(enabled=False)
session.set_dont_like(["naked", "nsfw"])
session.set_do_follow(True, percentage=70, times=1)
session.follow_by_tags(['dancestudioownerlife', 'dancestudioowner', 'danceteacher'], amount=30)
session.follow_user_followers(['twinklestardance', 'confettionthedancefloor', 'toddlerandpreschoolballet', 'studiotogodance', 'leapnlearn'],
                              amount=20, randomize=True, sleep_delay=60, interact=False)
session.follow_likers(['twinklestardance', 'confettionthedancefloor', 'toddlerandpreschoolballet', 'studiotogodance', 'leapnlearn'],
                      photos_grab_amount=3, follow_likers_per_photo=5, randomize=True, sleep_delay=60, interact=False)
session.follow_commenters(['twinklestardance', 'confettionthedancefloor', 'toddlerandpreschoolballet', 'studiotogodance', 'leapnlearn'],
                      amount=3, daysold=180, max_pic=20, sleep_delay=60, interact=False)
session.unfollow_users(amount=50, nonFollowers=True, style='RANDOM', unfollow_after=432000, sleep_delay=60)

session.end()


little_swan_dance_follow_bot.py
from dotenv import load_dotenv

load_dotenv()

from instapy import InstaPy
from instapy import smart_run
from instapy import set_workspace
from os import getenv

import schedule
import time

set_workspace(path=getenv('LITTLE_SWAN_DANCE_WORKSPACE'))


def job():
    session = InstaPy(username=getenv('LITTLE_SWAN_DANCE_USERNAME'), password=getenv('LITTLE_SWAN_DANCE_PASSWORD'),
                      headless_browser=True).login()
    session.set_quota_supervisor(enabled=True,
                                 sleep_after=["likes", "comments_d", "follows", "unfollows", "server_calls_h"],
                                 sleepyhead=True, stochastic_flow=True, notify_me=False,
                                 peak_likes_hourly=20,
                                 peak_likes_daily=100,
                                 peak_comments_hourly=12,
                                 peak_comments_daily=180,
                                 peak_follows_hourly=20,
                                 peak_follows_daily=150,
                                 peak_unfollows_hourly=10,
                                 peak_unfollows_daily=50,
                                 peak_server_calls_hourly=None,
                                 peak_server_calls_daily=4000)

    session.set_use_clarifai(enabled=False, api_key=getenv('CLARIFAI_API_KEY'))

    with smart_run(session):
        session.set_do_like(enabled=True)
        session.set_do_comment(enabled=False)
        session.set_dont_like(["naked", "nsfw"])
        session.set_do_follow(True, percentage=70, times=1)
        session.set_user_interact(amount=1, randomize=True, percentage=100, media='Photo')
        session.follow_by_tags(['dancestudioownerlife', 'dancestudioowner', 'danceteacher'], amount=30)
        session.follow_user_followers(
            ['twinklestardance', 'confettionthedancefloor', 'toddlerandpreschoolballet', 'studiotogodance',
             'leapnlearn'],
            amount=20, randomize=True, sleep_delay=30, interact=True)
        session.follow_likers(
            ['twinklestardance', 'confettionthedancefloor', 'toddlerandpreschoolballet', 'studiotogodance',
             'leapnlearn'],
            photos_grab_amount=3, follow_likers_per_photo=5, randomize=True, sleep_delay=30, interact=True)
        session.follow_commenters(
            ['twinklestardance', 'confettionthedancefloor', 'toddlerandpreschoolballet', 'studiotogodance',
             'leapnlearn'],
            amount=3, daysold=180, max_pic=20, sleep_delay=30, interact=True)
        session.unfollow_users(amount=50, nonFollowers=True, style='RANDOM', unfollow_after=432000, sleep_delay=30)

        # session.end()


schedule.every().day.at("11:50").do(job)

while True:
    schedule.run_pending()
    time.sleep(10)


little_swan_dance_follow_bot-app.py
from dotenv import load_dotenv
load_dotenv()

from instapy import InstaPy
from os import getenv

set_workspace(path=getenv('LITTLE_SWAN_DANCE_WORKSPACE'))

session = InstaPy(username=getenv('LITTLE_SWAN_DANCE_USERNAME'), password=getenv('LITTLE_SWAN_DANCE_PASSWORD'), headless_browser=True).login()
session.set_quota_supervisor(enabled=True, sleep_after=["likes", "comments_d", "follows", "unfollows", "server_calls_h"], sleepyhead=True, stochastic_flow=True, notify_me=False,
                              peak_likes_hourly=20,
                              peak_likes_daily=100,
                               peak_comments_hourly=12,
                               peak_comments_daily=180,
                                peak_follows_hourly=20,
                                peak_follows_daily=150,
                                 peak_unfollows_hourly=10,
                                 peak_unfollows_daily=50,
                                  peak_server_calls_hourly=None,
                                  peak_server_calls_daily=4000)

session.set_use_clarifai(enabled=True, api_key=getenv('CLARIFAI_API_KEY'))
session.set_do_like(enabled=False)
session.set_do_comment(enabled=False)
session.set_dont_like(["naked", "nsfw"])
session.set_do_follow(True, percentage=70, times=1)
session.follow_by_tags(['dancestudioownerlife', 'dancestudioowner', 'danceteacher'], amount=30)
session.follow_user_followers(['twinklestardance', 'confettionthedancefloor', 'toddlerandpreschoolballet', 'studiotogodance', 'leapnlearn'],
                              amount=20, randomize=True, sleep_delay=60, interact=False)
session.follow_likers(['twinklestardance', 'confettionthedancefloor', 'toddlerandpreschoolballet', 'studiotogodance', 'leapnlearn'],
                      photos_grab_amount=3, follow_likers_per_photo=5, randomize=True, sleep_delay=60, interact=False)
session.follow_commenters(['twinklestardance', 'confettionthedancefloor', 'toddlerandpreschoolballet', 'studiotogodance', 'leapnlearn'],
                      amount=3, daysold=180, max_pic=20, sleep_delay=60, interact=False)
session.unfollow_users(amount=50, nonFollowers=True, style='RANDOM', unfollow_after=432000, sleep_delay=60)

session.end()


leo_june_follow_bot.py
from dotenv import load_dotenv

load_dotenv()

from instapy import InstaPy
from instapy import smart_run
from instapy import set_workspace
from os import getenv

import schedule
import time

set_workspace(path=getenv('LEO_JUNE_WORKSPACE'))


def job():
    session = InstaPy(username=getenv('LEO_JUNE_USERNAME'), password=getenv('LEO_JUNE_PASSWORD'),
                      headless_browser=False).login()
    session.set_quota_supervisor(enabled=True,
                                 sleep_after=["likes", "comments_d", "follows", "unfollows", "server_calls_h"],
                                 sleepyhead=True, stochastic_flow=True, notify_me=False,
                                 peak_likes_hourly=20,
                                 peak_likes_daily=100,
                                 peak_comments_hourly=12,
                                 peak_comments_daily=180,
                                 peak_follows_hourly=20,
                                 peak_follows_daily=150,
                                 peak_unfollows_hourly=10,
                                 peak_unfollows_daily=50,
                                 peak_server_calls_hourly=None,
                                 peak_server_calls_daily=4000)

    session.set_use_clarifai(enabled=False, api_key=getenv('CLARIFAI_API_KEY'))

    with smart_run(session):
        session.set_do_like(enabled=True)
        session.set_do_comment(enabled=False)
        session.set_dont_like(["naked", "nsfw", "explicit"])
        session.set_do_follow(True, percentage=70, times=1)
        # session.follow_by_tags(['toddlerlife', 'toddlerstyle', 'boymom', 'momlife', 'babyboy', 'littlefashionista', 'totsytle', 'instakidstyle', 'childrenfashion', 'stylishkids', 'instakids', 'toddlerfashion', 'kidslookbook'], amount=30)
        session.set_user_interact(amount=1, randomize=True, percentage=100, media='Photo')
        session.follow_user_followers(
            ['Alexandnovakids', 'PatPat_clothing', 'shoplittlemango', 'Harlowandoakley_',
             'gracefulwillows', 'summer_n_may', 'shopbombon', 'dolcebellakids', 'mangokids', 'aliceandames', 'tatumandreesebaby'],
            amount=5, randomize=True, sleep_delay=30, interact=True)
        # session.follow_likers(
        #     ['Alexandnovakids', 'PatPat_clothing', 'shoplittlemango', 'Harlowandoakley_',
        #      'gracefulwillows', 'summer_n_may', 'shopbombon', 'dolcebellakids', 'mangokids', 'aliceandames',
        #      'tatumandreesebaby'],
        #     photos_grab_amount=1, follow_likers_per_photo=2, randomize=True, sleep_delay=30, interact=True)
        # session.follow_commenters(
        #     ['Alexandnovakids', 'PatPat_clothing', 'shoplittlemango', 'Harlowandoakley_',
        #      'gracefulwillows', 'summer_n_may', 'shopbombon', 'dolcebellakids', 'mangokids', 'aliceandames',
        #      'tatumandreesebaby'],
        #     amount=3, daysold=180, max_pic=5, sleep_delay=30, interact=True)
        session.unfollow_users(amount=100, nonFollowers=True, style='RANDOM', unfollow_after=172800, sleep_delay=30)

        # session.end()


schedule.every().day.at("21:29").do(job)

while True:
    schedule.run_pending()
    time.sleep(10)
